File: Bicicletas/src/Bicicletas/FactoriaMontana.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Bicicletas;

/**
 *
 * @author ravolk
 */
public class FactoriaMontana implements FactoriaCarreraYBicicleta{
    
    
    @Override
    public CarreraMontana crearCarrera()
    {
        return new CarreraMontana();
    }
    
    @Override 
    public BicicletaMontana crearBicicleta()
    {
        return new BicicletaMontana(int id);
    }
    
}

File: Bicicletas/src/Bicicletas/factoria_montana.rb

require_relative 'FactoriaCarreraYBicicleta'
require_relative 'carrera_montana'
require_relative 'BicicletaMontana'

class FactoriaMontana

  def crearCarrera
    return CarreraMontana.new
  end

  def crearBicicleta
    return BicicletaMontana.new
  end

end
